### ISSUE TITLE
Cannot compile

### ISSUE THREAD

**Issue by pegasos1** [cmr/openblas-provider#5]
When trying to compile blas/lapack crates, dependency openblas-provider is not compiling. I'm on Ubuntu 14.04. I've installed most dependencies that I can think of: liblapack-dev, libblas-dev, libopenblas-dev, gfortran. I've installed openblas from source. Compilation of openblas-provider keeps failing with the following stderr: 

```
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: warning: -jN forced in submake: disabling jobserver mode.
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

BFD: ../../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

BFD: ../../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../../libopenblas_haswellp-r0.2.14.a(ctbsv_CUN.o): Recognised but unhandled machine type (0x0) in Import Library Format archive
BFD: ../../../libopenblas_haswellp-r0.2.14.a(c_abs.o):2: Unexpected character `\016' in S-record file

make[1]: warning: -jN forced in submake: disabling jobserver mode.
../libopenblas_haswellp-r0.2.14.a: member ../libopenblas_haswellp-r0.2.14.a(saxpy.o) in archive is not an object
collect2: error: ld returned 1 exit status
make[1]: *** [../libopenblas_haswellp-r0.2.14.so] Error 1
make: *** [shared] Error 2
thread '<main>' panicked at '`"make" "libs" "netlib" "shared" "YES_CBLAS=1" "-j4"` failed: exit code: 2', ~/.cargo/registry/src/github.com-0a35038f75765ae4/openblas-provider-0.1.6/build.rs:57
```

**Comment by IvanUkhov** [comment]
The main purpose of this package is to make sure that you have OpenBLAS installed. If you don’t, the package will try to compile OpenBLAS from scratch, and the only prerequisite in this case is gfortran. Now, currently [there is no](https://github.com/xianyi/OpenBLAS/issues/598) a reliable way of figuring out whether OpenBLAS is installed. As a result, the package always compiles OpenBLAS, and this is exactly what you see above. Apparently something goes wrong in your environment. [Here](https://travis-ci.org/stainless-steel/blas) you can see how a successful compilation looks like, and you can try to compare Travis’ environment with what you have.

If you have successfully installed OpenBLAS yourself, you can try to make use of the system-openblas feature of the package as it is described in issue #4.

Please let me know if you find out what the problem is in your case. Thanks.


Regards,
Ivan

**Comment by ssgrn** [comment]
Hi Ivan,

Thanks so much for your help. After looking at that successful compilation, I realized I had Rust nightly. I switched to stable and everything compiled properly.

**Comment by IvanUkhov** [comment]
That’s strange. Rust’s version shouldn’t matter as the actual compilation is delegated to a Makefile and friends. [Here](https://travis-ci.org/stainless-steel/blas) I’m testing that crate on stable, beta, and nightly, and it compiles in all the cases. I think the problem was due to something else.

As an aside, I’ve just updated and published a new version of openblas-provider, v0.1.7, which compiles the latest version of OpenBLAS, v0.2.15.

Regards,
Ivan

**Comment by IvanUkhov** [comment]
I close this as it seems to be no longer an issue.

Best,
Ivan
